presto-hudi/src/main/java/com/facebook/presto/hudi/HudiPageSource.java

```diff
@@ -98,8 +111,19 @@ public HudiPageSource(
                 delegateIndex++;
             }
             outputIndex++;
+
+            // create type convert
+            if (!columnCoercions.isEmpty() &&
+                    columnCoercions.containsKey(column.getName()) &&
+                    !column.getHiveType().equals(columnCoercions.get(column.getName()))) {
+                coercersBuilder.add(Optional.of(HiveCoercer.createCoercer(typeManager, columnCoercions.get(column.getName()), column.getHiveType())));
```

Comment:
columnCoercions has the new HiveType of columns after doing schema evolution. Should we reverse the last 2 variables in this call? The method signature goes like this - `static HiveCoercer createCoercer(TypeManager typeManager, HiveType fromHiveType, HiveType toHiveType)`. That is why thinking maybe `column.getHiveType()` should be the second parameter in this call? Please correct me if I am wrong.

Reply:
```
static HiveCoercer createCoercer(TypeManager typeManager, HiveType fromHiveType, HiveType toHiveType)
```
column.getHiveType() is column type from hive metastore, In theory, it is the latest schema
columnCoercions.get(column.getName())  return a old hive type before DDL

Reply:
I see. Thanks for clarifying that.